Lint error: 'options' is not defined

**Describe the bug**

When building the latest version of master, gulp fails with a lint error.

```
[11:02:32] Using gulpfile /****/chromeos-filesystem-onedrive/gulpfile.js
[11:02:32] Starting 'default'...
[11:02:32] Starting 'clean'...
[11:02:32] Finished 'clean' after 5.07 ms
[11:02:32] Starting 'lint'...
[11:02:33]
/****/chromeos-filesystem-onedrive/src/scripts/onedrive_client.js
  623:13  error  'options' is not defined  no-undef
  624:42  error  'options' is not defined  no-undef
  625:29  error  'options' is not defined  no-undef

✖ 3 problems (3 errors, 0 warnings)

[11:02:33] 'lint' errored after 712 ms
[11:02:33] ESLintError in plugin "gulp-eslint"
Message:
    Failed with 3 errors
Details:
    domainEmitter: [object Object]
    domain: [object Object]
    domainThrown: false

[11:02:33] 'default' errored after 721 ms
```

**To Reproduce**
Steps to reproduce the behaviour:
1. Check out the latest version of master.
2. `npm install`
3. `gulp`

**Additional Notes**
If you disable the lint step in the gulp file, it does build & run ok (sort of).

Looking at the lines in the file it is referring to, for the first one in the file at-least, the `options` object hasn't been initialised at the point you are setting a value inside it; is it just a case of initialising it first, or was a pre-filled object supposed to be passed?

--- a/src/scripts/onedrive_client.js
+++ b/src/scripts/onedrive_client.js
@@ -923,20 +923,17 @@ class OneDriveClient {
         var data = JSON.stringify({
             path: path
         });
+        var splitPath = path.split("/");
         console.log('operation is below')
         switch(operation){
             case 'create_folder':
                 console.log('making a directory');
                 data = JSON.stringify({
-                    path: path,
+                    name: splitPath.pop(),
                     folder: {}
                 });
-                if (path !== "") {
-                    //url += ":" + path + ":";
-                    url += ":" + path;
-                }
-                //url += "/children";
-                operation = 'PUT';
+                url += ":/" + splitPath.join("/") + ":/children";
+                operation = 'POST';
                 break;
             case 'delete':
                 console.log('deleting a file');

--- a/src/scripts/onedrive_client.js
+++ b/src/scripts/onedrive_client.js
@@ -518,11 +518,11 @@ class OneDriveClient {
 
     writeFile(filePath, data, offset, openRequestId, successCallback, errorCallback) {
         const writeRequest = this.writeRequestMap[openRequestId];
-        if (writeRequest.uploadId) {
+        if (writeRequest.uploadUrl) {
             this.doWriteFile(filePath, data, offset, openRequestId, writeRequest, successCallback, errorCallback);
         } else {
-            this.startUploadSession(sessionId => {
-                writeRequest.uploadId = sessionId;
+            this.startUploadSession(filePath, sessionUrl => {
+                writeRequest.uploadUrl = sessionUrl;
                 this.doWriteFile(filePath, data, offset, openRequestId, writeRequest, successCallback, errorCallback);
             }, errorCallback);
         }
@@ -543,7 +543,7 @@ class OneDriveClient {
             dataType: 'binary',
             responseType: 'arraybuffer'
         }, data, data => {
-            this.startUploadSession(sessionId => {
+            this.startUploadSession(filePath, sessionUrl => {
                 if (length < data.byteLength) {
                     // Truncate
                     const req = {
@@ -551,7 +551,7 @@ class OneDriveClient {
                         data: data.slice(0, length),
                         offset: 0,
                         sentBytes: 0,
-                        uploadId: sessionId,
+                        uploadUrl: sessionUrl,
                         hasMore: true,
                         needCommit: true,
                         openRequestId: null
@@ -569,7 +569,7 @@ class OneDriveClient {
                             data: reader.result,
                             offset: 0,
                             sentBytes: 0,
-                            uploadId: sessionId,
+                            uploadUrl: sessionUrl,
                             hasMore: true,
                             needCommit: true,
                             openRequestId: null
@@ -597,7 +597,7 @@ class OneDriveClient {
             data: data,
             offset: offset,
             sentBytes: 0,
-            uploadId: writeRequest.uploadId,
+            uploadUrl: writeRequest.uploadUrl,
             hasMore: true,
             needCommit: false,
             openRequestId: openRequestId
@@ -614,15 +614,15 @@ class OneDriveClient {
             extPattern.test(metadata.name);
     }
         
-    startUploadSession(successCallback, errorCallback) {
-        const data = this.jsonStringify({
+    startUploadSession(filePath, successCallback, errorCallback) {
+        const reqData = this.jsonStringify({
             close: false
         });
         console.log('STARTINGUPLOADSESSION');
         console.log(this);
         new HttpFetcher(this, 'startUploadSession', {
             type: 'POST',
-            url: "https://graph.microsoft.com/v1.0/me/drive/items/" + this + "/createUploadSession",
+            url: "https://graph.microsoft.com/v1.0/me/drive/root:/" + filePath + ":/createUploadSession",
             headers: {
                 'Authorization': 'Bearer ' + this.access_token_,
                 'Content-Type': 'application/json'
@@ -631,12 +631,10 @@ class OneDriveClient {
                 "@odata.type": "microsoft.graph.driveItemUploadableProperties",
                 "@microsoft.graph.conflictBehavior": "replace"
             },
-        }, data, result => {
+        }, reqData, result => {
             console.log('creating upload session');
             console.log(result);
-            options.uploadUrl = result.uploadUrl;
-            this.sendContents.call(this, options, successCallback, errorCallback);
-            successCallback(options);
+            successCallback(result.uploadUrl);
         }, errorCallback).fetch();
     }
 

--- a/src/scripts/onedrive_client.js
+++ b/src/scripts/onedrive_client.js
@@ -446,21 +446,35 @@ class OneDriveClient {
 
     readFile(filePath, offset, length, successCallback, errorCallback) {
         const data = JSON.stringify({path: filePath});
+        if (offset > 0) {
+            console.log("readFile:: Offset reads are not currently supported");
+            errorCallback();
+            return;
+        }
         const range = 'bytes=' + offset + '-' + (offset + length - 1);
         new HttpFetcher(this, 'readFile', {
             type: 'GET',
-            url: "https://graph.microsoft.com/v1.0/me/drive/root:" + filePath + ":/content",
+            url: "https://graph.microsoft.com/v1.0/me/drive/root:" + filePath + "?select=id,@microsoft.graph.downloadUrl",
             headers: {
                 'Authorization': 'Bearer ' + this.access_token_,
-                'Range': range
-            },
-            dataType: 'binary',
-            responseType: 'arraybuffer'
+            }
         }, {
             data: data,
             range: range
         }, result => {
-            successCallback(result, false);
+            console.log("starting download")
+            console.log(result);
+            new HttpFetcher(this, 'readFile', {
+                type: 'GET',
+                url: result["@microsoft.graph.downloadUrl"],
+                dataType: 'binary',
+                responseType: 'arraybuffer'
+            }, {
+                data: data,
+                range: range
+            }, result2 => {
+                successCallback(result2, false);
+            }, errorCallback).fetch();
         }, errorCallback).fetch();
     }
 

